.github/workflows/lint-test-build.yml

```diff
@@ -0,0 +1,128 @@
+name: Lint / Test / Build
+
+on:
+  pull_request:
+
+
+env:
+  DEFAULT_NODE_VERSION: 16.x
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+
+  build:
+    name: ${{ matrix.action }} (${{ matrix.node-version }})
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [16.x, 18.x]
+        action: [build, lint, test]
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Cache node modules
+        id: cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            node_modules
+          key: ${{ runner.os }}-node-${{ matrix.node-version }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ matrix.node-version }}-
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v4
+        if: steps.cache.outputs.cache-hit != 'true'
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - run: npm ci
+        if: steps.cache.outputs.cache-hit != 'true'
+
+      - name: Set up QEMU
+        if: matrix.action == 'build'
+        id: qemu
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: amd64,arm64
+
+      - name: Set up Docker Buildx
+        id: buildx
+        if: matrix.action == 'build'
+        uses: docker/setup-buildx-action@v3
+        with:
+          install: true
+
+      - name: Cache Docker layers
+        id: cache-docker
+        if: steps.buildx.outcome == 'success'
+        uses: actions/cache@v4
+        with:
+          path: ${{ runner.temp }}/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ matrix.node-version }}-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-${{ matrix.node-version }}-
+
+      - name: Docker cache size fix
+        if: steps.cache-docker.outcome == 'success'
+        run: |
+          mv -v ${{ runner.temp }}/.buildx-cache ${{ runner.temp }}/.buildx-cache-old || true
```

Comment:
Could you please provide a reference for why this fix is necessary, so that we can ensure that we remove it when this workaround is no longer necessary?

Reply:
Thank you. I'm thinking of there being a comment to explain any step is not immediately obvious, so that as this file is adjusted for maintenance that they can be reviewed to see if the assumptions are still correct. Ideally pointing to an issue tracker if it is something that has been raised for the given action.